src/include/dawg/rate.h

```diff
@@ -24,20 +24,22 @@ class rate_model {
 	bool create(const std::string &rname, It first, It last) {
 		static std::string name_keys[] = {
 			std::string("const"),
-			std::string("gamma-invariant")
+			std::string("gamma-invariant"),
+            std::string("zero")
 		};
 		switch(key_switch(rname, name_keys)) {
 			case 0:
-				return create_const(first, last);
+				return create_const();
 			case 1:
 				return create_gamma(first, last);
+            case 2:
+                return create_zero();
 		};
 		return DAWG_ERROR("Invalid rate model; no model named '" << rname << "'");		
 		
 	}
-	
-	template<typename It>
-	inline bool create_const(It first, It last) {
+
+	inline bool create_const() {
 		name_ = "const";
```

Comment:
Yes.

--
Reed A. Cartwright, PhD
Barrett Honors Faculty
Assistant Professor of Genomics, Evolution, and Bioinformatics
School of Life Sciences
Human and Comparative Genomics Laboratory
The Biodesign Institute
Arizona State University
==================
Availability: http://links.asu.edu/CartwrightCalendar
Address: The Biodesign Institute, PO Box 875301, Tempe, AZ 85287-5301 USA
Packages: The Biodesign Institute, 1001 S. McAllister Ave, Tempe, AZ
85287-5301 USA
Office: Biodesign A-224A, 1-480-965-9949
Website: http://cartwrig.ht/

On Thu, Jul 28, 2016 at 2:51 PM, Zach <notifications@github.com> wrote:

> In src/include/dawg/rate.h
> <https://github.com/reedacartwright/dawg/pull/9#discussion_r72709843>:
>
> >  		};
> >  		return DAWG_ERROR("Invalid rate model; no model named '" << rname << "'");		
> >  		
> >  	}
> > -	
> > -	template<typename It>
> > -	inline bool create_const(It first, It last) {
> > +
> > +	inline bool create_const() {
> >  		name_ = "const";
>
> Should this apply to the create_zero() function as well?
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/reedacartwright/dawg/pull/9/files/7e190686646511a5c3536db994ba09b8dfeef613#r72709843>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAGOHrX6sWPfdxuYsmMOZFGOKNSxtt6cks5qaSRqgaJpZM4JXqZp>
> .
>